app/Http/Controllers/Admin/PostController.php

```diff
@@ -25,28 +26,67 @@ public function index()
     public function show_posts()
     {
         $posts = $this->postRepo->getAllOrderByDesc();
-        return view('admin.posts.posts', ['posts' => $posts]);
+        return view('admin.posts.posts', ['posts' => $posts, 'title' => 'Posts Management', 'index_nav' => Self::INDEX_NAV]);
     }
 
     public function add_post_page()
     {
-        return view('admin.posts.add_post');
+        return view('admin.posts.add_post', ['title' => 'Add Post', 'index_nav' => Self::INDEX_NAV]);
     }
 
     public function add_post(StorePostRequest $request)
     {
+        $data = $request->all();
         // Save file on public/images
         $imageName = Carbon::now()->timestamp . '_' . $request->file('image')->getClientOriginalName();
         $request->image->move(public_path('images'), $imageName);
         // Insert data on table posts
         $urlImage = 'images/' . $imageName;
         $this->postRepo->create([
             'user_id' => auth()->guard('admin')->user()->id,
-            'title' => $request->input('title'),
-            'description' => $request->input('description'),
+            'title' => $data['title'],
+            'description' => $data['description'],
             'image' => $urlImage
         ]);
+        return redirect()->route('posts');
+    }
+
+    public function edit_post_page($post_id)
+    {
+        $post = $this->postRepo->find($post_id);
+        if ($post->exists()) {
+            return view('admin.posts.edit_post', ['post' => $post, 'title' => 'Edit Post', 'index_nav' => Self::INDEX_NAV]);
+        }
+    }
 
+    public function edit_post(StorePostRequest $request)
+    {
+        $data = $request->all();
+        if (!isset($data['image'])) {
```

Comment:
this->postRepo->update đang giống nhau, nên gom lại e nhé
ở đây chỉ khác là có data image thì thêm biến image thôi, e có thể xử lý vd như
```
$dataUpdate = [
                    'user_id' => auth()->guard('admin')->user()->id,
                    'title' => $data['title'],
                    'description' => $data['description'],
                ];
```
nếu check có image thì $dataUpdate['image'] =  $urlImage
postRepo update thì để ngoài cùng, không cần else
```
$this->postRepo->update(
                $data['id'],
                $dataUpdate
            );

```